R/dummy_functions.R

```diff
@@ -36,30 +37,46 @@ example_module <- function(label = "example teal module", datanames = "all", tra
           )
         })
 
+        table_data <- reactive({
+          within(data(),
+            {
+              table <- dataname
+            },
+            dataname = as.name(input$dataname)
+          )
+        })
+
+        table_data_decorated <- srv_transform_data("decorate", data = table_data, transforms = decorators)
+
         output$text <- renderPrint({
-          req(input$dataname)
-          data()[[input$dataname]]
+          req(table_data_decorated)
+          table_data_decorated()[["table"]]
         })
 
         teal.widgets::verbatim_popup_srv(
           id = "rcode",
-          verbatim_content = reactive(teal.code::get_code(data())),
+          verbatim_content = reactive(teal.code::get_code(req(table_data_decorated()))),
```

Comment:
Here we go. Until this moment modules was (kind of) safe from errors. If a module was made in a good way so that it doesn't throw any validation errors, this line should never make any problems. Now, we are giving app user to interfere with module "internal" code and thus also fail it. "trigger on success" doesn't work here, because here for example we have `renderPrint`, which generically doesn't escape when `table_data_decorate` is failing initially. All these mean that we can no longer expect only `teal_data` or `qenv.error`, because it could be also a `shiny.error`.
This is a serious dilemma because `qenv.error` doesn't make sense at all now. It looks like we don't need `qenv.error` anymore, we just need a `req` statements. Are we?

Reply:
Let's give it some time to marinate in our heads